src/enclave/main.cpp

```diff
@@ -1,12 +1,14 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the Apache 2.0 License.
+#include "common/enclave_interface_types.h"
 #include "ds/json.h"
 #include "ds/logger.h"
 #include "ds/spin_lock.h"
 #include "ds/stacktrace_utils.h"
 #include "enclave.h"
 #include "enclave_time.h"
 #include "oe_shim.h"
+#include "version.h"
```

Comment:
Just a note, but don't think its actually a problem at the moment.

Most of our directory-less includes are in the same directory as the current file. This is an exception, since its in `build/generated`. This is not installed, so is not visible to an app. Its unclear that this is different and may confuse us when it breaks in future - perhaps we should put these all under `build/_gen/generated`, so we can call them `generated/version.h` here (but keeping an extra layer so we're adding `build/_gen` to the include dirs, not `build/`).

This won't break at the moment because apps don't actually build this file (its built into `libccf`, and they link to that). But now that it _looks_ like we have a `ccf_version` that can be used within the enclave, its probably only a matter of time before we try to use that in an app, and find that the include is surprisingly unavailable.

I'm not sure what we'd do to fix this. Maybe I'm saying `version.h` should be put under `include/ccf` as a public header, even though its generated?

Reply:
This a very good point, I think distributing `version.h` as a public include is an excellent idea.

Reply:
See #2562